### ISSUE TITLE
Cannot require 'google/cloud/datastore'

### ISSUE THREAD

**Issue by welkie** [GoogleCloudPlatform/google-cloud-ruby#1226]
I'm having an issue using Datastore in a local (soon to go on Heroku) Rails app. I add it to my Gemfile with:

```ruby
gem 'google-cloud-datastore'
```

But when I try to create a Datastore object:

```ruby
datastore = Google::Cloud::Datastore.new
```

I get:

```ruby
NoMethodError: undefined method `new' for Google::Cloud::Datastore:Module
```

I did notice in the documentation that there is a require statement, so when I modifed my Gemfile:

```ruby
gem 'google-cloud-datastore', require 'google/cloud/datastore'
```

I get this when launching the rails console:

```ruby
/usr/bin/ruby -e $stdout.sync=true;$stderr.sync=true;load($0=ARGV.shift) /home/mwelke/Projects/GcdTester/bin/rails console development
/var/lib/gems/2.3.0/gems/google-cloud-datastore-0.23.0/lib/google/cloud/datastore/v1/datastore_client.rb:28:in `require': cannot load such file -- google/gax (LoadError)
```

Note that I added the GOOGLE_CLOUD_PROJECT and GOOGLE_CLOUD_KEYFILE_JSON environment variables, so that I could ommit the authentication information as I use the Datastore API.

**Comment by blowmage** [comment]
Hi @welkie, you need to use the require to call the `new` method. The LoadError is a know issue with the file permissions in the google-gax library. The gem's files are only readable by the user that was used to install the gem. A fix is to adjust the gem's file permissions. You can do this by finding the paths that the `google-gax` and `googleapis-common-protos` gems are installed in and to fix the file permissions.

```
$ bundle show google-gax
$ find <GOOGLE-GAX-PATH> -type f -perm 640 -print -exec chmod 644 {} \;
```

We are working on updating all gems to use the newer google-gax version that fixes the file permissions.

Here are some related issues:

https://github.com/googleapis/gax-ruby/issues/54
https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/1193
https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/1194

**Comment by welkie** [comment]
@blowmage Thanks for replying. I actually solved this by trying again at creating a Rails app to test this. I was able to instantiate a Datastore object after successfully requring Datestore. The first app I created, I had a lot of "skip" flags because I thought that less would be better. I was only testing Google Cloud Platform, nothing else:

```ruby
rails new GcdTester --skip-active-record --skip-sprockets --skip-action-mailer --skip-javascript --skip-turbolinks --skip-test
```

My new app, I only skipped ActiveRecord, and it worked:

```ruby
2.3.3 :001 > datastore = Google::Cloud::Datastore.new
 => #<Google::Cloud::Datastore::Dataset:0x000000030e75b0 @service=Google::Cloud::Datastore::Service(gcdtester-158404)>
```

Very happy to be ready to tinker. But I'm curious now what caused the issue I experienced.

**Comment by blowmage** [comment]
> Very happy to be ready to tinker. 

Us too. Let us know how we can help.

> But I'm curious now what caused the issue I experienced.

You likely didn't have the LoadError because you installed the google-gax gem with the same user you were running the app as.

**Comment by blowmage** [comment]
Hi @welkie, today we released new gems that reply on updated dependencies that fix this issue. Can you upgrade to the latest gems and verify that this fixes this issue or not? Thanks!

**Comment by welkie** [comment]
Works well for me today. Unfortunately, I'm not sure I even understand the issue I originally had. Your posts inferred that it was an issue with my permissions as I installed and used gems. A user mismatch. I thought it had to do with having so many skip switches in my ```rails new```.

Today, I used the exact same ```rails new``` as in my post above where I encountered the problem, with all the skip switches, and I had in my Gemfile:

```ruby
gem 'google-cloud-datastore', require: 'google/cloud/datastore'
```

And I'm using the following class:

```ruby
class Gcd
  def self.get_instance
    @instance ||= Google::Cloud::Datastore.new
  end
end
```

to attempt to make a GCD client the following way:

```
irb(main):001:0> Gcd.get_instance
=> #<Google::Cloud::Datastore::Dataset:0x000000047aff68 @service=Google::Cloud::Datastore::Service(gcdtester-158404)>
```

and it seems to work perfectly now. :) Thanks for following up.
